### PR TITLE
Dead Water and core's Storm Trident: Change "mermen" to "merfolk"

### PR DESCRIPTION
**Pull request by stevecotton**
* Dead Water: Change "mermen" to "merfolk" (part of #2940)
    This commit excludes changes to S05 Tirigaz, I think changing the dialogue
    there is going to be more complicated, and it's better for that to have a
    separate PR.

* Prose: the Storm Trident's description uses "merfolk" (issue #2936)
    Also update Dead Water's copy of it to use race= instead of a
    list of unit types (as done for the core object in 613dd431).

### CODE REVIEW

**Comment by stevecotton** [comment]
@beetlenaut, please could I request a review of this?

File: data/campaigns/Dead_Water/units/Citizen.cfg [hunk 1]
```diff
@@ -14,7 +14,7 @@
     advances_to=Merman Brawler,Merman Fighter,Merman Hunter
     cost=9
     usage=fighter
-    description= _ "Mermen are mostly peaceful, but will fight to defend their homes. In this situation, even mermen with no weapons or training can be dangerous."
+    description= _ "Merfolk are mostly peaceful, but will fight to defend their homes. In this situation, even merfolk with no weapons or training can be dangerous."
```
Comment (stevecotton) [inline data/campaigns/Dead_Water/units/Citizen.cfg]: I've changed the Citizen, but left the corresponding text in the Brawler as male, it seemed to fit better to me.

**Comment by stevecotton** [comment]
Changelogs: I think this doesn't need campaign-specific entries, just a general "Updated prose referring to the merfolk" in either "Language and i18n" or "Miscellaneous".

**Comment by beetlenaut** [comment]
This is all fine with me.

**Comment by sevu** [comment]
For me half of the changes sound strange, as merfolk refers to the race, while mermen can refer to both the race or a group of people. It may mean more tweaks than a simple swapping put of word, such as adding articles or rewording.
But I'm not a native speaker, I may be wrong.

**Comment by stevecotton** [comment]
I'm assuming that "merfolk" can be used for a group as well as a race in Wesnoth, it would be good to have a few more reviewers comment.

Although I'm a native speaker, I've been living in a different country for years - my brain now automatically handles bad English without noticing.

**Comment by CelticMinstrel** [comment]
Seems fine to me to use "merfolk" to refer to a group? I don't understand how there could be a problem with it.

**Comment by stevecotton** [comment]
Sorry if you've just received an update notification. I added a one-line additional commit, and then 10 seconds later force-pushed to change it back to the original branch.

The user_team_name is currently "Merman Forces", and I thought "Merfolk" would be better than "Merfolk Forces" (compared with other campaigns, where the team is often simply "Rebels" or similar). And then I realised it would need to be "team_name^Merfolk".

**Pull request merged**